File: Fortran/ex5/trainLinearReg.f90
	subroutine trainLinearReg(M, N, X, y, lambda, theta)
	implicit none
	
	integer::M,  N
	double precision,dimension(M)::y
	double precision,dimension(M, N)::X
	double precision,dimension(N)::theta, grad
	double precision::lambda, Jcost 
	integer::i,j
	
	theta = 0	
	call fmincg(M, N, theta, X, y,lambda, 200);
	
	end subroutine trainLinearReg

File: C/ex5/trainLinearReg.c
double linearRegCostFunction(int M,int N,double X[M][N], double y[M], double theta[N], double lambda, double grad[N]);
int fmincg(double (*linearRegCostFunction)(int M,int N,double X[M][N], double y[M],double theta[N], double lambda, double grad[N]),
	int M,int N, double nn_params[N],double X[M][N],double y[M],double lambda, int maxCostCalls);

int trainLinearReg(int M, int N, double X[M][N],double y[M], double lambda, double theta[N])
{
	int i;
	
	for(i=0;i<N;i++){ theta[i] = 0;}
		
	fmincg(linearRegCostFunction, M, N, theta, X, y,lambda, 200);
	
	return 0;
}
